src/App.tsx

```diff
@@ -1,61 +1,143 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User;
+}
+
+function getUserById(userId: number): User | undefined {
+  return usersFromServer.find(user => user.id === userId);
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(
+    todosFromServer.map(todo => ({
+      ...todo,
+      user: getUserById(todo.userId),
+    })),
+  );
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(true);
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(true);
+
+  const getNextId = () => {
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const validTitle = event.target.value.replace(/[^a-zA-Zа-яА-Я0-9 ]/g, '');
+
+    setTitle(validTitle);
+    if (validTitle.trim()) {
+      setHasTitleError(false);
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    if (+event.target.value !== 0) {
```

Comment:
The condition `+event.target.value !== 0` is used to determine if a user is selected. Ensure that the default option has a value of `0` to correctly trigger this condition.